### ISSUE TITLE
Permission denied on calculation item... because it references a secured object

### ISSUE THREAD

**Issue by KayceeInMelbourne** [TabularEditor/TabularEditor#1089]
Hi there, 

I've just introduced a calculated item to conditionally format a column in Power BI.  

The first user to be granted access to the new report set cannot see the data in the report as a result of an error "Permission denied on calculation item 'x' because it references a secured object". It works fine for those with admin access.

To my knowledge, only RLS has been applied (no OLS - certainly not intentionally). This user has access to everything and as such their role has no conditions set. The Role is blank against every table therefore implicitly allowing access to everything.

Until now, this has worked perfectly.  Can you pls advise what I can look at in my model to resolve this or alternatively documentation that will assist. 

Cheers
Kaycee

**Comment by otykier** [comment]
The error message indicates that the model does in fact have OLS applied. So I recommend that you take a look at each table permission within the relevant role, to see if there is any OLS applied:

<img width="662" alt="image" src="https://user-images.githubusercontent.com/8976200/231372558-62115103-ceb7-4fa2-8262-b188cd5d0cf1.png">

**Comment by KayceeInMelbourne** [comment]
Brilliant, thank you for the quick reply.

To confirm, "default" is what it should be but "None" or any other option would imply OLS?

**Comment by otykier** [comment]
Correct

**Comment by KayceeInMelbourne** [comment]
@otykier ,

Thanks again for your advice. Unfortunately this hasn't resolved the issue.  I've popped in a mock up role for testing. 

As you can see, everything says default.  I also took a quick look at every table and every column of every table and they all say 'Default' on the OLS. 

Are there any other areas that could be affecting this?

![image](https://user-images.githubusercontent.com/117255785/231606209-5e2fe9c7-8ae0-4521-84f9-7969cd3aceaf.png)

**Comment by otykier** [comment]
What about the model permission on the role itself?

<img width="570" alt="image" src="https://user-images.githubusercontent.com/8976200/231668129-aba22ad3-bf3c-45ca-963a-3a74e231bebe.png">

**Comment by KayceeInMelbourne** [comment]
@otykier , it also states "Read" as your image says above. 

For reference, I have currently removed all Roles so that my client can see the data while we continue through development and test. 

When I re-add the role, the same issue occurs.  I thought this might be useful to know in that it isn't being 'added' to the role. It is either happening by default, or its a setting somewhere else that remains even when I have no roles in the model.

**Comment by KayceeInMelbourne** [comment]
@otykier just floating this back to the top.  I see it's been marked as closed but as yet I've not been able to use security roles due to this issue.  

Would you be able to reopen and advise if there is anything else I can consider that might be triggering an OLS issue?

**Comment by otykier** [comment]
Did you assign the user in question to the role? (As soon as a role is created on a dataset, a user will either need admin permission, or be a member of one of the roles on the dataset, before they are able to see anything)

https://docs.tabulareditor.com/te3/tutorials/data-security/data-security-about.html#3-assign-users-to-roles

**Comment by KayceeInMelbourne** [comment]
@otykier. Yes, it definitely was assigned.  

In an attempt to narrow down the object causing the issue, I simplified the code in the offending calculation group until the error disappeared.  

As it turns out, there code used the ISSELECTEDMEASURE function.  2 of the measures in that function had been deleted. Instead of offering an error, it worked fine until RLS was applied in which case the OLS error was returned.  

I assume this was because the object (measure) didn't exist.  The error was unfortunately misleading.  

My apologies that I have used your time on this - it was frustrating for me so I can imagine it isn't ideal for you.  

Thank you though for your support. It's improved my understanding of an area of PBI/Tabular Editor I didn't realise existed.
